http-endpoint-example/host.py

```diff
@@ -0,0 +1,108 @@
+import argparse
+import json
+import os.path
+# pprint used to make the output easier to read
+import pprint
+
+from flask import Flask, request
+import requests
+
+
+# Iniatilise global variables so they can be used in the handled messages
+DOWNLOAD_DIR = 'objects'
+app = Flask(__name__)
+args = []
+checked_diagnostics = ()
+start_time = 0
+end_time = 0
+
+
+def get_object(url):
+    if not os.path.exists(DOWNLOAD_DIR):
+        os.mkdir(DOWNLOAD_DIR)
+    target_path = os.path.join(DOWNLOAD_DIR, url.split('/')[-1])
+    headers = {'x-api-key': args.api_key}
+    print('Beginning download of {} to {}'.format(url, target_path))
+    r = requests.get(url, headers=headers, stream=True)
+    with open(target_path, 'wb') as fd:
+        for chunk in r.iter_content(chunk_size=1024):
+            fd.write(chunk)
+    print('Completed download of {} to {}'.format(url, target_path))
+
+
+def handle_message(message):
+    print(message['metadata']['name'])
+    if message['metadata']['name'] in checked_diagnostics:
+        forecast_period = int(message['metadata']['forecast_period'])
+        if start_time <= forecast_period <= end_time:
+            get_object(message['url'])
+
+
+@app.route('/', methods=['GET', 'POST', 'PUT'])
+def sns():
+    # AWS sends JSON with text/plain mimetype
+    js = json.loads(request.data)
+    if args.verbose:
+        pprint.pprint(js)
+    message_type = request.headers.get('x-amz-sns-message-type')
+
+    '''
+    If being used in a production there should be a step here to
```

Comment:
"If being used in a production environment" or "in production"? :police_car: :christmas_tree: